cpp/core/memory/MemoryAllocator.cc

```diff
@@ -157,7 +157,7 @@ bool StdMemoryAllocator::reallocate(void* p, int64_t size, int64_t newSize, void
 }
 
 bool StdMemoryAllocator::reallocateAligned(void* p, uint64_t alignment, int64_t size, int64_t newSize, void** out) {
-  if (newSize <= 0) {
+  if (newSize <= 0 || p == nullptr) {
```

Comment:
Can we throw exceptions (and other changes as well)? Since it's not usual to have a nullptr passed in for such functions.

Reply:
Could refer to the macros in https://github.com/apache/incubator-gluten/blob/6d71d27884dd6bda38848cb012a63fb37ee2d298/cpp/core/utils/Exception.h#L42 if needed